src/evm/decoder.rs

```diff
@@ -44,6 +44,21 @@ type RegistryFn =
     dyn Fn(ComponentWithState, Header, Arc<RwLock<DecoderState>>) -> DecodeFut + Send + Sync;
 type FilterFn = fn(&ComponentWithState) -> bool;
 
+/// A builder for creating protocol streams, decoding blocks, and managing protocol state updates.
```

Comment:
This is not a builder? Also please note that this is a private class (public only to the parent module).